test/e2e/specs/editor/various/rich-text.spec.js

```diff
@@ -0,0 +1,615 @@
+		await page.keyboard.type( '.' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should not highlight more than one format', async ( {
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+		await page.keyboard.type( '1' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+		await page.keyboard.type( ' 2' );
+		await pageUtils.pressKeyWithModifier( 'shift', 'ArrowLeft' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+
+		const count = await page
+			.locator( '*[data-rich-text-format-boundary]' )
+			.count();
+		expect( count ).toBe( 1 );
+	} );
+
+	test( 'should return focus when pressing formatting button', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Some ' );
+		await editor.showBlockToolbar();
+		await page.click( '[aria-label="Bold"]' );
+		await page.keyboard.type( 'bold' );
+		await editor.showBlockToolbar();
+		await page.click( '[aria-label="Bold"]' );
+		await page.keyboard.type( '.' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should transform backtick to code', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'A `backtick`' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'z' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should undo backtick transform with backspace', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '`a`' );
+		await page.keyboard.press( 'Backspace' );
+
+		// Expect "`a`" to be restored.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should not undo backtick transform with backspace after typing', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '`a`' );
+		await page.keyboard.type( 'b' );
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.press( 'Backspace' );
+
+		// Expect "a" to be deleted.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'should not undo backtick transform with backspace after selection change', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '`a`' );
+		await page.evaluate( () => new Promise( window.requestIdleCallback ) );
+		// Move inside format boundary.
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.press( 'Backspace' );
+
+		// Expect "a" to be deleted.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
```

Comment:
The snapshot is empty, so I think this can compare against an empty string. Will need to delete the file too in addition to this change.

```suggestion
		expect( await editor.getEditedPostContent() ).toBe( '' );
```